### DISCUSSION TITLE
Using drag and drop on all the column and not only on the drag icon ?

### DISCUSSION THREAD

**Discussion by CamachoPt** [KevinVandy/material-react-table#272] [Q&A] [answered]
I am struggling to understand how to do this. 

Is it possible to remove the drag and drop icon and use all the column as draggable ?

**Comment by KevinVandy** [comment, marked as answer]
You can hide the the drag handles with `enableColumnDragging={false}` even if `enableColumnOrdering` is true.

It is possible to make a whole column draggable, but you will need to hook up all of the drag events yourself in `muiTableHeadCellProps` and/or `muiTableBodyCellProps`. Might not be as hard as you think.

You can take these two MRT source files for the drag handle icon buttons as inspiration:

https://github.com/KevinVandy/material-react-table/blob/main/packages/material-react-table/src/head/MRT_TableHeadCellGrabHandle.tsx

https://github.com/KevinVandy/material-react-table/blob/main/packages/material-react-table/src/buttons/MRT_GrabHandleButton.tsx
